src/cmd/build.js

```diff
@@ -60,8 +61,21 @@ function defaultPackageCreator(
           filter: (...args) => fileFilter.wantFile(...args),
         })
         .then((buffer) => {
+          let messageData: any;
+          if (manifestData.default_locale) {
+            messageData = readFileSync(
+               path.join(sourceDir, '_locales', manifestData.default_locale,
+                'messages.json'));
+          }
+          let extensionName: string;
+          if (messageData) {
+            extensionName = JSON.parse(messageData).extensionName.description;
```

Comment:
`name` can even be something like `__MSG_hello__foo __MSG_bar_baz__ qux` so using something like the following code should cover it.

```javascript
messageData = ...
manifestData.name = manifestData.name.replace(/__MSG_(.+?)__/g, function(matches) {
	if (messageData[matches[1]] && messageData[matches[1]].message) {
		return messageData[matches[1]].message;
	} else {
		return matches[0];
	}
});
```